src/common/telemetryClient.ts

```diff
@@ -4,14 +4,23 @@
 "use strict";
 import * as vscode from "vscode";
 import TelemetryReporter from "vscode-extension-telemetry";
+import { Utility } from "./utility";
 import { Constants } from "./constants";
 
 const packageJSON = vscode.extensions.getExtension(Constants.ExtensionId).packageJSON;
 const extensionVersion: string = packageJSON.version;
 const aiKey: string = packageJSON.aiKey;
+const isInternal: boolean = Utility.isInternalUser();
 
 export class TelemetryClient {
     public static sendEvent(eventName: string, properties?: { [key: string]: string; }): void {
+        if (properties) {
+            properties[Constants.isInternalPropertyName] = isInternal === true ? 'true' : 'false';

```

Comment:
isInternal?'true':'false' ?

Reply:
use === to avoid implicit conversion in the comparison, though this should not happen in the isinternal scenario. Just want to make sure the comparison is robust if there's any changes on how to initialize isInternal value in the future.